resourcename/rangeparents.go

```diff
@@ -0,0 +1,23 @@
+package resourcename
+
+// RangeParents iterates over all parents of the provided resource name.
+// The iteration order is from root ancestor down to the closest parent.
+// For full resource names, the service is omitted.
```

Comment:
What is the reason for including collections as well as resources?

Reply:
So that we don't have to solve the problem of determining what is a collection and not, since we don't assume that we have access to the full metadata about the pattern here.

Reply:
Gotcha!